api/types/database.go

```diff
@@ -521,22 +521,26 @@ func (d *DatabaseV3) GetIAMAction() string {
 // GetIAMResources returns AWS IAM resources that provide access to the database.
 func (d *DatabaseV3) GetIAMResources() []string {
 	aws := d.GetAWS()
+	partition := "aws"
+	if aws.Region != "" && (aws.Region == "cn-northwest-1" || aws.Region == "cn-north-1") {
+		partition = "aws-cn"
+	}
```

Comment:
Could you add a helper function somewhere here like `GetPartitionFromRegion`, or use the existing function like `IsCNRegion` instead of comparing each region?

https://github.com/gravitational/teleport/blob/18558b88e94e92cb43c74f4b61a4a53769f72211/api/utils/aws/region.go#L25-L34

thanks